cardano-api/src/Cardano/Api/TxBody.hs

```diff
@@ -1433,6 +1545,13 @@ data TxValidityLowerBound era where
 deriving instance Eq   (TxValidityLowerBound era)
 deriving instance Show (TxValidityLowerBound era)
 
+instance Semigroup (TxValidityLowerBound era) where
+    TxValidityNoLowerBound <> x = x
+    x <> TxValidityNoLowerBound = x
+    (TxValidityLowerBound wit l) <> (TxValidityLowerBound _ r) = TxValidityLowerBound wit (max l r)
```

Comment:
I disagree with using `max` here. `SlotNo` has a `Num` instance, I think we should add the values together with `(+)`. That way a user can adjust the validity lower bound as they see fit, with positive or negative values.